core/common/src/ECSqlReader.ts

```diff
@@ -364,15 +364,15 @@ export class ECSqlReader implements AsyncIterableIterator<QueryRowProxy> {
   }
 
   /**
-   *
+   * @internal
    */
-  private async runWithRetry(request: DbQueryRequest) {
-    const needRetry = (rs: DbQueryResponse) => (rs.status === DbResponseStatus.Partial || rs.status === DbResponseStatus.QueueFull || rs.status === DbResponseStatus.Timeout) && (rs.data.length === 0);
+  protected async runWithRetry(request: DbQueryRequest) {
+    const needRetry = (rs: DbQueryResponse) => (rs.status === DbResponseStatus.Partial || rs.status === DbResponseStatus.QueueFull || rs.status === DbResponseStatus.Timeout) && (typeof(rs.data) === "undefined" || rs.data.length === 0);
```

Comment:
```suggestion
    const needRetry = (rs: DbQueryResponse) => (rs.status === DbResponseStatus.Partial || rs.status === DbResponseStatus.QueueFull || rs.status === DbResponseStatus.Timeout) && !rs?.data.length;
```
Just a suggestion to simplify long condition.
If you don't like that, consider replacing `typeof(rs.data) === "undefined"` with `rs.data === undefined`.